models/errors.go

```diff
@@ -15,15 +15,16 @@ func (err simpleError) Error() string {
 	return err.message
 }
 
+// NewSimpleError returns the custom defined error of type SimpleError.
 func NewSimpleError(msg string) simpleError {
```

Comment:
Correct, I just realized that this could have been done as 

```
func NewConversionError(msg string) ConversionError {
	return ConversionError{simpleError{msg}}
}
```

I've removed NewSimpleError(..) @tsmaeder 

The blackbox test works!
Could you please review?